internal/gatewayapi/securitypolicy.go

```diff
@@ -589,7 +589,18 @@ func (t *Translator) buildOIDC(
 	}
 
 	// Generate a unique cookie suffix for oauth filters
-	suffix := utils.Digest32(string(policy.UID))
+	// We use the digest of the redirect URL to generate the cookie suffix so that
+	// multiple OIDC configurations with the same redirect URL will have the same
+	// cookie suffix. This allows us to share the same cookies across multiple
+	// oauth filters at multiple routes. Once users log in at one route, they will
+	// be able to access other routes with the same OIDC redirect URL (the same
+	// OIDC clientID) without logging in again.
+	//
+	// zhaohuabing: This is a workaround to allow multiple routes to share the same OIDC configuration.
+	// We should change this back to use policy UID after Gateway API supports
+	// targeting a policy to multiple routes.
+	// See https://github.com/kubernetes-sigs/gateway-api/discussions/2927#discussioncomment-8991869
+	suffix := utils.Digest32(*oidc.RedirectURL)
```

Comment:
The `oidc` is not present in every security policy and the `redirectURL` might be missing

Reply:
You mean `RedirectURL` may be null? We should use `ClientID` here.